src/program/program.service.ts

```diff
@@ -474,18 +491,23 @@ export class ProgramService {
 	//Mutations
 
 	/// Create a new module
-	async addModule(data: Prisma.ModuleCreateInput) {
+	async addModule(data: NewModule) {
 		//find out if there is a duplicate user
 		const get = await this.prisma.module.findMany({
 			where: {
 				moduleNumber: data.moduleNumber
 			}
 		});
+
+		const create = Prisma.validator<Prisma.ModuleCreateInput>()({
```

Comment:
This might need to move into the else block, and validate the where clause of the get `const` separately before using it.